src/states/joining_peer.rs

```diff
@@ -32,9 +32,7 @@ use std::{
 };
 
 /// Time after which bootstrap is cancelled (and possibly retried).
-pub const JOIN_TIMEOUT: Duration = Duration::from_secs(120);
```

Comment:
`This happens because the `completed_events` container in the chain accumulator is not shared state (maybe it should be?)` as you found out this is not ok. Does this patch still make sense/should we revert it too?

Reply:
I decided to remove the join attempts for two reasons:
1. It was the trigger for the failure mode described in the commit message
2. It was no longer necessary, as the churn serializatin should take care of the issue that it was originally meant to fix. That is, the join attempts were a workaround for a situation where a `JoinRequest` got ignored due to ongoing churn in the target section. With churn serialization this should not happen anymore and so this workaround is not necessary and would only add additional complexity to the code

Reply:
OK, I'm on board with the idea of simplifying. I'm not 100% `no longer necessary` is correct.
(Probably need to fix commit comment because it mention something that no longer is part of the history).

Reply:
> I'm not 100% no longer necessary is correct.

I'm not 100% either, but pretty close. Previously what could happen is that a joining node would send `JoinRequest` which the section received and voted `Online`. But due to other ongoing churn this vote might have never accumulated. The join attempts were a workaround for this, so the joining node would send the `JoinRequest` again, hoping that the section has stabilized enough to be able to handle it. With churn serialization, my impression is this **should** not be necessary. Or can you think of a case where the join attempts would still be needed?

One issue is that it can take some time for the section to respond to the joining node (due to churn backlogging) which is why I increased the join timeout.

> Probably need to fix commit comment because it mention something that no longer is part of the history

Not sure what you mean by this. What is no longer part of the history?

Reply:
> This happens because the completed_events container in the chain accumulator is not shared state (maybe it should be?)

Reply:
(keep on Ctrl Enter to make new line)... I thought you you removed that commit that kept completed_events and made them not shared state.

Reply:
Hm, after some time thinking about this, I think you are right. The comment is outdated. Also this whole commit is not really needed. The failure happened because of not resetting the completed_events cache on every parsec reset while also not making it part of shared state. As I removed that commit, that particular failure mode should not be possible anymore. I think this commit is still useful to simplify things, but not essential anymore. So what do you think: should I remove the commit or just edit the commit message?

Reply:
I leave you to decide. Whatever makes getting all soak test passing (for mock at least) fastest.